Automattic/sensei
# Fix global button colors in Learning Mode for the Course theme

Partially fixes #6582.

NOTE: I'm waiting on designs for what the buttons should look like when a lesson is completed, but I'll address it in a separate PR. I'd like to get this merged as it might affect other cards.

## Proposed Changes
Fix global button colors not working with the Course theme by updating the markup and CSS classes for the _Take Quiz_ and _Complete Lesson_ buttons so that they match what the buttons block uses. This PR also updates the button text in the editor from _Complete Course_ to _Complete Lesson_.

Note that this PR does not attempt to:
- Fix any existing editor UI issues with Divi. It was already pretty broken before this PR (at least prior to WP 6.2 when the site editor was accessible).
- Make global button colors work for all block based themes. That will be a bigger problem to solve.

Other important points worth noting:
- Since Divi and Astra are not block themes, global styles are not expected to work for either of them.
- The only theme where global button colors will work is Course.
- Global button colors are only applied to buttons with the _Fill_ style (i.e. the _Take Quiz_ button). I basically matched the same behaviour as when setting global button colors for fill and outline buttons added to a post.

## Testing Instructions


1. Check out [this branch](https://github.com/Automattic/themes/pull/6990) of the Course theme.
2. Activate the Course theme.
3. Open the Lesson template in the site editor.
4. Set global button colors for both the background and text.
5. Ensure the Take Quiz button colors are updated in the editor.
6. Check a lesson with a quiz on the frontend and ensure the Take Quiz button colors match the global styles.
7. Switch to the Divi theme (on WP 6.0?).
8. Check that the editor is visually the same as before this PR (see screenshot below).
9. Check a lesson with a quiz on the frontend and ensure it looks the same as before this PR.
10. Repeat steps 7 - 9 for the Astra theme.

## Screenshots

### Course Theme - Editor
![course-editor-after](https://user-images.githubusercontent.com/1190420/224421434-97992ca4-7db8-492c-a6d3-d4c2db9e2696.jpg)

### Course Theme - Frontend
![course-frontend-after](https://user-images.githubusercontent.com/1190420/224421496-cbf85891-6409-4105-9f66-a1f34327c3c5.jpg)

### Divi - Editor
![divi-editor-after](https://user-images.githubusercontent.com/1190420/224421579-7090b45a-3d29-46eb-b7ce-39949b243612.jpg)

### Divi - Frontend
![divi-frontend](https://user-images.githubusercontent.com/1190420/224421588-148713d1-13a1-4825-9e8e-99f4bf050ad0.jpg)

### Astra - Editor
![astra-editor-after](https://user-images.githubusercontent.com/1190420/224421641-66e499ef-0460-46a3-84b1-a8a41449c03f.jpg)

### Astra - Frontend
![astra-frontend](https://user-images.githubusercontent.com/1190420/224421638-074010df-e0de-4a66-9cba-2cc672b35d76.jpg)


## Pre-Merge Checklist

- [x] Known issues are created as new GitHub issues* (global button colors not working with block themes, editor UI issues in Divi)
